Add documentation

weiznich opened the pull request:
This PR adds documentation to the crate. It also removes any item that does not need to be public from the public API. In addition it fixes some issues with how the dependencies are specified and remove the default feature set.

I propose to do a first release after merging this PR

pgab commented on src/oracle/connection/mod.rs:
```diff
@@ -347,8 +459,8 @@ impl OciConnection {
                     OciDataType::Double => oracle::sql_type::OracleType::BinaryDouble,
-                    OciDataType::Text => oracle::sql_type::OracleType::NVarchar2(2_000_000),
-                    OciDataType::Binary => oracle::sql_type::OracleType::Raw(2_000_000),
+                    OciDataType::Text => oracle::sql_type::OracleType::Varchar2(2000),
```
Not sure, why, we change the type and size here?

pgab commented on src/oracle/connection/mod.rs:
```diff
@@ -360,8 +472,8 @@ impl OciConnection {
         }
-
-        stmt.execute_named(&binds).map_err(ErrorHelper::from)?;
+        //        dbg!(stmt.oci_attr::<oracle::oci_attr::Statement>());
```
Is this some leftover?

MrMuetze replied:
Can this be removed?

pgab commented on src/oracle/connection/mod.rs:
```diff
@@ -360,8 +472,8 @@ impl OciConnection {
-
-        stmt.execute_named(&binds).map_err(ErrorHelper::from)?;
+        //        dbg!(stmt.oci_attr::<oracle::oci_attr::Statement>());
+        dbg!(stmt.execute_named(&binds)).map_err(ErrorHelper::from)?;
```
Do we want `dbg!` here at all times?

pgab commented on src/test/mod.rs:
```diff
@@ -2394,5 +2394,72 @@ fn batch_insert_1() {
+}
+
+// #[test]
+// fn explicit_single_returning() {
```
Do we want this test or not? If the first, it should be included, otherwise remove it completely.

MrMuetze commented on src/test/mod.rs:
```diff
@@ -2394,5 +2394,72 @@ fn batch_insert_1() {
     assert_eq!(res[1].big, Some(-3));
 }
 
+#[test]
```
Are both of these tests supposed to stay like that?

MrMuetze reviewed (changes requested):
Just two questions remain. All the comments and docs look good!

weiznich commented:
All of what was commented above was not supposed to be in this PR at all. Seems like I added that in the last commit that was only supposed to fix the CI. I've just rewrote the last commit so that's all excluded now.
